app/src/main/java/com/example/androidtrainingproject/ui/shared/AppTopBar.kt

```diff
@@ -12,20 +14,29 @@ import androidx.compose.material3.TopAppBarDefaults
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.style.TextOverflow
+import androidx.navigation.NavController
 import com.example.androidtrainingproject.R
 import com.example.androidtrainingproject.ui.theme.Black
 import com.example.androidtrainingproject.ui.theme.White
 
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun AppTopBar(middleText: String) {
+fun AppTopBar(
+    middleText: String,
+    navController: NavController,
+    backIcon: Boolean? = false,
```

Comment:
What's the point in booleans being optional? Isn't false the same thing in this context? Moreover, you're already default initializing them to `false` which makes it impossible for them to be `null` anyway.